File: frontend/app/models/timeRegistration.ts
// src/types/TimeRegistration.ts

export interface TimeRegistrationCreateDTO {
    projectId: string;       // Guid is mapped to string in TypeScript
    freelancerId: string;    // Guid is mapped to string in TypeScript
    workDate: string;        // DateOnly is mapped to string in TypeScript (ISO 8601 format)
    hoursWorked: number;
    description: string;
}
export interface TimeRegistrationDTO {
    id: string;
    projectId: string;       // Guid is mapped to string in TypeScript
    freelancerId: string;    // Guid is mapped to string in TypeScript
    workDate: string;        // DateOnly is mapped to string in TypeScript (ISO 8601 format)
    hoursWorked: number;
    description: string;
    projectName: string;
}

  

File: backend/src/Core/Entities/TimeRegistration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Entities
{
    public class TimeRegistration
    {
        public Guid Id { get; set; }
        public Guid ProjectId { get; set; }
        public Project Project { get; set; }
        public Guid FreelancerId { get; set; }
        public DateOnly WorkDate { get; set; }
        public decimal HoursWorked { get; set; }
        public string Description { get; set; }
        

    }
}
